---
title: "Re: Riak issue after power outage"
description: ""
project: community
lastmod: 2017-06-01T05:27:49-07:00
sitemap:
  priority: 0.2
layout: mailinglistitem
mailinglist_id: "msg18257"
mailinglist_parent_id: "msg18255"
author_name: "Fred Dushin"
project_section: "mailinglistitem"
sent_date: 2017-06-01T05:27:49-07:00
---


There may be better ways, but what I would do is:

Stop Riak, blow away your ${platform\_data\_dir}/anti\_entropy directory (and 
yz\_anti\_entropy, if you have search enabled), and restart. This will force a 
rebuild of your hash trees. They rebuild automatically anyway; this is just 
making them rebuild faster.

So it would recover .. eventually.

-Fred

> On Jun 1, 2017, at 7:13 AM, Guido Medina  wrote:
> 
> Correction: I replied to an old e-mail instead of creating a new one and 
> forgot to change the subject.
> 
> Hi all,
> 
> My impression of Riak has always been that it would recover from anything but 
> yesterday we had the worst happened, there was a power outage so all the 
> servers in a Riak cluster went down, once they were back we have been having 
> these constant so my questions are:
> How can I recover from this?
> Isn't Riak supposed to restore corrupted hashes from other nodes?
> 
>> 2017-06-01 12:07:43.795 [info] 
>> <0.659.0>@riak\_kv\_vnode:maybe\_create\_hashtrees:234 
>> riak\_kv/1164634117248063262943561351070788031288321245184: unable to start 
>> index\_hashtree: {error,{{badmatch,{error,{db\_open,"Corruption: truncated 
>> record at end of 
>> file"}}},[{hashtree,new\_segment\_store,2,[{file,"src/hashtree.erl"},{line,725}]},{hashtree,new,2,[{file,"src/hashtree.erl"},{line,246}]},{riak\_kv\_index\_hashtree,do\_new\_tree,3,[{file,"src/riak\_kv\_index\_hashtree.erl"},{line,712}]},{lists,foldl,3,[{file,"lists.erl"},{line,1248}]},{riak\_kv\_index\_hashtree,init\_trees,3,[{file,"src/riak\_kv\_index\_hashtree.erl"},{line,565}]},{riak\_kv\_index\_hashtree,init,1,[{file,"src/riak\_kv\_index\_hashtree.erl"},{line,308}]},{gen\_server,init\_it,6,[{file,"gen\_server.erl"},{line,304}]},{proc\_lib,init\_p\_do\_apply,3,[{file,"proc\_lib.erl"},{line,239}]}]}}
>> 2017-06-01 12:07:45.457 [error] <0.27505.112> CRASH REPORT Process 
>> <0.27505.112> with 0 neighbours exited with reason: no match of right hand 
>> value {error,{db\_open,"Corruption: truncated record at end of file"}} in 
>> hashtree:new\_segment\_store/2 line 725 in gen\_server:init\_it/6 line 328
> This is not our production cluster but our development is now stuck because 
> all the data we had there would take days to restore from another 
> cluster, I hope there is a solvable problem.
> 
> Regards,
> 
> Guido.
> \_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
> riak-users mailing list
> riak-users@lists.basho.com
> http://lists.basho.com/mailman/listinfo/riak-users\_lists.basho.com

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
riak-users mailing list
riak-users@lists.basho.com
http://lists.basho.com/mailman/listinfo/riak-users\_lists.basho.com

